community/kernel-api/src/main/java/org/neo4j/internal/kernel/api/security/AccessMode.java

```diff
@@ -50,6 +51,15 @@ public AuthorizationViolationException onViolation( String msg )
                     }
                 },
 
+        NOT_IN_TRANSACTION( false, false, false, false, false, false )
```

Comment:
It feels a bit awkward to have the security context validate that we are in a transaction. E.g.
```
        if ( !accessMode.allowsReads() )
        {
            throw accessMode.onViolation( ... );
        }
```
will throw `NotInTransactionException`. I think that this should be handeled separately, would `ktx.assertOpen()` work?

Reply:
Yes, but that would lead to double `assertOpen()` calls in the happy path, which I would guess could slow things down. This change was mostly to make development nicer so we could get more descriptive error messages while writing new code. If we think the solution is architecturally awkward I would rather revert it than slow down production code.

Personally I think this solution is non-obvious, but it still makes sense to me when I look at it. Eg. if we are not in transaction we are not allowed any operations, and the best exception which explains that to offending code is a `NotInTranscationException`.